eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -107,16 +107,20 @@ case class Commitments(channelVersion: ChannelVersion,
     if (localParams.isFunder) {
       // The funder always pays the on-chain fees, so we must subtract that from the amount we can send.
       val commitFees = commitTxFeeMsat(remoteParams.dustLimit, reduced)
-      // the funder needs to keep an extra reserve to be able to handle fee increase without getting the channel stuck
-      // (see https://github.com/lightningnetwork/lightning-rfc/issues/728)
-      val funderFeeReserve = htlcOutputFee(2 * reduced.feeratePerKw)
-      val htlcFees = htlcOutputFee(reduced.feeratePerKw)
-      if (balanceNoFees - commitFees < offeredHtlcTrimThreshold(remoteParams.dustLimit, reduced)) {
+      // the funder needs to keep an extra buffer to be able to handle a x2 feerate increase and an additional htlc to avoid
+      // getting the channel stuck (see https://github.com/lightningnetwork/lightning-rfc/issues/728).
+      val funderFeeBuffer = commitTxFeeMsat(remoteParams.dustLimit, reduced.copy(feeratePerKw = 2 * reduced.feeratePerKw)) + htlcOutputFee(2 * reduced.feeratePerKw)
+      val amountToReserve = commitFees.max(funderFeeBuffer)
```

Comment:
How can `amountToReserve` not equal `funderFeeBuffer`?

Reply:
Because of the trim limit (damn this is tricky)...
Imagine you have a commit tx with many HTLCs in it that are slightly above the trim limit when using `reduced.feeratePerKw`.
When you use `2 * reduced.feeratePerKw` all these HTLCs become trimmed: the feerate is bigger, but the commit tx becomes smaller. If the number of pending HTLCs was high enough, the total fee may end up smaller than before.

Reply:
Added an example that showcases this in the latest commit.